consensus/src/backup/saver.rs

```diff
@@ -1,22 +1,38 @@
 use std::io::Write;
 
 use crate::{units::UncheckedSignedUnit, Data, Hasher, Receiver, Sender, Signature, Terminator};
+use async_trait::async_trait;
 use codec::Encode;
 use futures::{FutureExt, StreamExt};
 use log::{debug, error};
 
 const LOG_TARGET: &str = "AlephBFT-backup-saver";
 
+#[async_trait]
+/// Write backups to peristent storage.
+pub trait BackupWriter {
+    /// Append new data to the backup.
+    async fn append(&mut self, data: &[u8]) -> std::io::Result<()>;
+}
+
+#[async_trait]
+impl<W: Write + Send> BackupWriter for W {
+    async fn append(&mut self, data: &[u8]) -> std::io::Result<()> {
+        Write::write_all(self, data)?;
+        self.flush()
+    }
```

Comment:
Doesn't this have to be in some wrapper that tells the async executor that it should be run on a thread that is allowed to block? Otherwise we only fix the problem in case downstream users don't use a writer but implement `BackupWriter` themselves. Maybe that's the best way to go anyway and we should just remove this default impl?

Reply:
see
https://github.com/Cardinal-Cryptography/AlephBFT/pull/399#issuecomment-1903601695 and https://github.com/Cardinal-Cryptography/AlephBFT/pull/399#issuecomment-1905695333

Reply:
Ok, so to avoid bigger changes we can keep it this way if it's only used for testing anyway.